code generators/de.fhdo.lemma.model_processing.code_generation.container_base/src/main/java/de/fhdo/lemma/model_processing/code_generation/container_base/intermediate_model_validation/DockerComposeFileIntermediateModelValidator.xtend

```diff
@@ -0,0 +1,51 @@
+package de.fhdo.lemma.model_processing.code_generation.container_base.intermediate_model_validation
+
+import de.fhdo.lemma.operation.intermediate.IntermediatePackage
+import de.fhdo.lemma.model_processing.annotations.IntermediateModelValidator
+import de.fhdo.lemma.model_processing.builtin_phases.intermediate_model_validation.AbstractIntermediateDeclarativeValidator
+import de.fhdo.lemma.model_processing.code_generation.container_base.util.Util
+import de.fhdo.lemma.operation.intermediate.IntermediateOperationNode
+import org.eclipse.xtext.validation.Check
+
+/**
+ * The container base model validator is responsible for checking general aspects for
+ * the container base code generator.
+ *
+ * @author Philip Wizenty(philip.wizenty@fh-dortmund.de)
+ */
+@IntermediateModelValidator
+class DockerComposeFileIntermediateModelValidator extends AbstractIntermediateDeclarativeValidator {
+
+    /*
+     * Get the namespace for the intermediate operation model
+     */
+    override getLanguageNamespace() {
+        return IntermediatePackage.eNS_URI
+    }
+
+    /*
+     * Check if the intermediate operation node contains an endpoint and if the port is
+     * set correctly.
+     */
+    @Check
+    def checkPortOfDeployedInfrastructureNode(IntermediateOperationNode node) {
+        if (node.endpoints === null || node.endpoints.empty) {
+            error("The endpoints are not defined, but are required by the code generator.",
+                IntermediatePackage.Literals.INTERMEDIATE_ENDPOINT_TECHNOLOGY__ENDPOINT
+            )
+        } else {
+            node.endpoints.forEach[e |
```

Comment:
I suggest to factor the else-branch out to a dedicated validation method for Endpoints only, i.e.,
```
@Check
def checkEndpointPort(IntermediateEndpoint) {
...
}
```

Then the else-branch is not necessary and you do not need to iterate at all.